* Put $all value for Required-(Start|Stop) in the LSB header.
* Change the "Default-Stop" field.
* Allow /etc/default/shinken to override the path of .ini files.
* "invoke-rc.d shinken status" returns 1 if at least one daemon has a problem.
--- a/bin/init.d/shinken
+++ b/bin/init.d/shinken
@@ -23,10 +23,10 @@
 
 ### BEGIN INIT INFO
 # Provides:          shinken
-# Required-Start:    $local_fs
-# Required-Stop:     $local_fs
+# Required-Start:    $all
+# Required-Stop:     $all
 # Default-Start:     2 3 4 5
-# Default-Stop:      S 0 1 6
+# Default-Stop:      0 1 6
 # Short-Description: Shinken monitoring daemon
 # Description:       Shinken is a monitoring tool composed of many separated modules:
 #     - arbiter     : the main one : control everything else.
@@ -290,7 +290,9 @@
 	output=$("$modfilepath" -d -c "$SKONFCFG" $DEBUGCMD 2>&1)
         rc=$?
     elif [ "$mod" != "arbiter" ]; then
-        output=$("$modfilepath" -d -c "$ETC/${mod}d.ini" $DEBUGCMD 2>&1)
+        modinifile_varname=$(echo ${mod}CFG | tr '[:lower:]' '[:upper:]')
+        modinitfile=$(eval echo '$'$modinifile_varname)
+        output=$("$modfilepath" -d -c "${modinifile}" $DEBUGCMD 2>&1)
         rc=$?
     else
         output=$("$modfilepath" -d -c "$SHINKENCFG" -c "$SHINKENSPECIFICCFG" $DEBUGCMD 2>&1)
@@ -517,10 +519,14 @@
 do_cmd_on() {
     action=$1
     mods=$2
+    local return_value
+    return_value=0
     for mod in $mods
     do
-        do_${action}_ "$mod"
+        # If at least one action fails, the return value is 1.
+        do_${action}_ "$mod" || return_value=1
     done
+    return $return_value
 }
 
 
